stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -83,53 +85,56 @@ internal class ChannelControllerImpl(
-    private val _endOfNewerMessages = MutableLiveData(false)
-    private val _endOfOlderMessages = MutableLiveData(false)
-    private val _loading = MutableLiveData(false)
-    private val _hidden = MutableLiveData(false)
-    private val _muted = MutableLiveData(false)
-    private val _watchers = MutableLiveData<Map<String, User>>(mapOf())
-    private val _members = MutableLiveData<Map<String, Member>>()
-    private val _loadingOlderMessages = MutableLiveData(false)
-    private val _loadingNewerMessages = MutableLiveData(false)
-    private val _channelData = MutableLiveData<ChannelData>()
-    private val _oldMessages = MutableLiveData<Map<String, Message>>()
+    private val _messages = MutableStateFlow<Map<String, Message>?>(null)
+    private val _watcherCount = MutableStateFlow<Int>(0)
+    private val _typing = MutableStateFlow<Map<String, ChatEvent>>(emptyMap())
+    private val _reads = MutableStateFlow<Map<String, ChannelUserRead>>(emptyMap())
+    private val _read = MutableStateFlow<ChannelUserRead?>(null)
+    private val _endOfNewerMessages = MutableStateFlow(false)
+    private val _endOfOlderMessages = MutableStateFlow(false)
+    private val _loading = MutableStateFlow(false)
+    private val _hidden = MutableStateFlow(false)
+    private val _muted = MutableStateFlow(false)
+    private val _watchers = MutableStateFlow<Map<String, User>>(emptyMap())
+    private val _members = MutableStateFlow<Map<String, Member>>(emptyMap())
+    private val _loadingOlderMessages = MutableStateFlow(false)
+    private val _loadingNewerMessages = MutableStateFlow(false)
+    private val _channelData = MutableStateFlow<ChannelData?>(null)
+    private val _oldMessages = MutableStateFlow<Map<String, Message>?>(emptyMap())
     internal var hideMessagesBefore: Date? = null
-    val unfilteredMessages: LiveData<List<Message>> =
-        Transformations.map(_messages) { it.values.toList() }
+    val unfilteredMessages: LiveData<List<Message>?> = _messages.map {
+        if (it == null) { null } else { it.values.toList() }
+    }.asLiveData()
 
     /** a list of messages sorted by message.createdAt */
     override val messages: LiveData<List<Message>> = messagesTransformation(_messages)
 
     override val oldMessages: LiveData<List<Message>> = messagesTransformation(_oldMessages)
 
-    private fun messagesTransformation(messages: MutableLiveData<Map<String, Message>>): LiveData<List<Message>> {
-        return Transformations.map(messages) { messageMap ->
-            messageMap.values
-                .asSequence()
-                .filter { it.parentId == null || it.showInChannel }
-                .filter { hideMessagesBefore == null || it.wasCreatedAfter(hideMessagesBefore) }
-                .sortedBy { it.createdAt ?: it.createdLocallyAt }
-                .toList()
-                .map { it.copy() }
+    private fun messagesTransformation(messages: MutableStateFlow<Map<String, Message>?>): LiveData<List<Message>> {
```

Comment:
Also is it good idea to use nullable type here? If so why messages and old messages live datas are non-nullable?